### ISSUE TITLE
Deterioration after "Completion window UX fixes" PR

### ISSUE THREAD

**Issue by kindermannhubert** [waf/CSharpRepl#148]
### Version

latest

### What happened?

When reading the description of @waf 's #137 I didn't realize the consequences of the first point. Space and equals sign characters are a vital part of completion commit characters. Their removal really hurts the behavior of other ordinary inputs. 

E.g. when writing:
```C#
class C { private readonly string X; }
```
my typical input is only following `cla C { pri re str X; }`.
Committing completion via the equals sign is similar.

Unfortunately, this is broken after this PR. The space character committing was one of the most important in my motivation to start using Roslyn's commit character set.

I totally understand that it needs improvements in other situations (like the one described in #137) but it has to be done in some context-aware manner. Commit characters should be removed only in annoying situations and not all.

We should probably check if we are using Roslyn's rules for character sets modification correctly - it's surprising to me that i's not already handled by Roslyn. If Roslyn can't provide the correct info, we need to detect (with the use of Roslyn) annoying situations by ourselves.

**Comment by waf** [comment]
Yeah, agreed. I've been struggling to get good behavior from the Roslyn commit characters -- as far as I can tell we're using it correctly, but the response from the roslyn api does not seem to be very sensitive to the current context.

It would not surprise me if I've customized it to the point where it "feels right to me" but doesn't match others' usages. I'm fine to revert https://github.com/waf/CSharpRepl/pull/137 or possibly just [7f2c801](https://github.com/waf/CSharpRepl/pull/137/commits/7f2c801c6efb2eec07d0149d26c05658a19ef35f).

**Comment by kindermannhubert** [comment]
And yeah, I get that everyone feels differently about "correct" behavior so I'd like to stick to VS behavior as much as possible.

**Comment by waf** [comment]
Reverted. Apologies for removing space, which was a major driver for the entire feature! Everyone uses VS a bit differently I guess. I agree that VS is the "gold standard" in behavior.
